api/src/chat/repositories/block.repository.ts

```diff
@@ -89,16 +89,112 @@ export class BlockRepository extends BaseRepository<
       Block,
       'findOneAndUpdate'
     >,
-    _criteria: TFilterQuery<Block>,
+    criteria: TFilterQuery<Block>,
     _updates:
       | UpdateWithAggregationPipeline
       | UpdateQuery<Document<Block, any, any>>,
   ): Promise<void> {
     const updates: BlockUpdateDto = _updates?.['$set'];
+    if (updates?.category) {
+      const movedBlockId = criteria._id;
+
+      // Find and update blocks that reference the moved block
+      await this.model.updateMany(
+        { nextBlocks: movedBlockId },
+        { $pull: { nextBlocks: movedBlockId } },
+      );
+
+      await this.model.updateMany(
+        { attachedBlock: movedBlockId },
+        { $set: { attachedBlock: null } },
+      );
+    }
 
     this.checkDeprecatedAttachmentUrl(updates);
   }
 
+  /**
+   * Pre-processing logic for updating blocks.
+   *
+   * @param query - The query to update blocks.
+   * @param criteria - The filter criteria for the update query.
+   * @param updates - The update data.
+   */
+  async preUpdateMany(
+    _query: Query<
```

Comment:
```suggestion
    query: Query<
```
Underscore prefix should be used only when the arg will be unused.